desktopexporter/internal/store/store.go

```diff
@@ -1,134 +1,288 @@
 package store
 
 import (
-	"container/list"
 	"context"
+	"database/sql"
+	"encoding/json"
 	"fmt"
+	"log"
 	"sync"
+	"time"
+
+	"github.com/marcboeker/go-duckdb"
 
 	"github.com/CtrlSpice/otel-desktop-viewer/desktopexporter/internal/telemetry"
 )
 
 type Store struct {
-	MaxQueueSize int
-	mut          sync.Mutex
-	TraceQueue   *list.List
-	TraceMap     map[string]telemetry.TraceData
+	mut       sync.Mutex
+	db        *sql.DB
+	connector *duckdb.Connector
 }
 
-func NewStore(maxQueueSize int) *Store {
+func NewStore() *Store {
+	c, err := duckdb.NewConnector("", nil)
+	if err != nil {
+		log.Fatalf("could not initialize new connector: %s", err.Error())
+	}
+
+	db := sql.OpenDB(c)
+	_, err = db.Exec(ENABLE_JSON)
+	if err != nil {
+		log.Fatalf("could not enable json: %s", err.Error())
+	}
+
+	_, err = db.Exec(CREATE_SPANS_TABLE)
+	if err != nil {
+		log.Fatalf("could not create table spans: %s", err.Error())
+	}
+
 	return &Store{
-		MaxQueueSize: maxQueueSize,
-		mut:          sync.Mutex{},
-		TraceQueue:   list.New(),
-		TraceMap:     map[string]telemetry.TraceData{},
+		mut:       sync.Mutex{},
+		db:        db,
+		connector: c,
 	}
 }
 
-func (store *Store) Add(_ context.Context, spanData telemetry.SpanData) {
-	store.mut.Lock()
-	defer store.mut.Unlock()
+func (s *Store) AddSpans(ctx context.Context, spans []telemetry.SpanData) {
+	s.mut.Lock()
+	defer s.mut.Unlock()
 
-	// Enqueue, then append, as the enqueue process checks if the traceID is already in the map to keep the trace alive
-	store.enqueueTrace(spanData.TraceID)
-	traceData, traceExists := store.TraceMap[spanData.TraceID]
-	if !traceExists {
-		traceData = telemetry.TraceData{
-			TraceID: spanData.TraceID,
-			Spans:   []telemetry.SpanData{},
+	con, err := s.connector.Connect(ctx)
```

Comment:
I wonder if we should store the connection in the `Store` so we don't have reconnect each time? Probably not significant at the scales this will be working at though, especially with an in-memory db

Reply:
Oh okay, so here the answer is yes and. 

I should store the connection per best practices ([https://duckdb.org/docs/guides/performance/how_to_tune_workloads.html#best-practices-for-using-connections](https://duckdb.org/docs/guides/performance/how_to_tune_workloads.html#best-practices-for-using-connections)), but also I have a `driver.Conn` for adding spans using the DuckDB appender and a `sql.Conn` and I need to figure out how to resolve that.